[WFLY-13089] MicroProfile REST Client 1.4 support

asoldano opened the pull request.

honza-kasik commented on microprofile/WFLY-13089-microprofile-rest-client-1.4.adoc:
```diff
@@ -0,0 +1,54 @@
+
+=== Test Plan
+
+Build and run RESTEasy testsuite, which pulls the MP REST Client TCK in a dedicated testsuite submodule.
```
Can you please provide links to the RESTEasy testsuite and MP REST Client TCK?

asoldano replied:
https://github.com/resteasy/Resteasy/tree/3.11/testsuite

It's automatically run as part of the RESTEasy build anyway (including the Travis CI on GitHub).

honza-kasik commented on microprofile/WFLY-13089-microprofile-rest-client-1.4.adoc:
```diff
@@ -0,0 +1,54 @@
+
+* RESTEasy provides an implementation of Microprofile REST Client 1.4.0 version
+** The only relevant change comparing to version 1.3.x is the requirement to support CDI injection in ClientHeadersFactory class
+** The optional JAXRS injection (@Context) support in ClientHeadersFactory class is not going to provided
```
Typo:

> is not going to provided

=>

> is not going to be provided

asoldano replied:
Fixed, thanks

honza-kasik commented on microprofile/WFLY-13089-microprofile-rest-client-1.4.adoc:
```diff
@@ -0,0 +1,54 @@
+== Requirements
+
+* RESTEasy provides an implementation of Microprofile REST Client 1.4.0 version
+** The only relevant change comparing to version 1.3.x is the requirement to support CDI injection in ClientHeadersFactory class
```
What are the criteria for a "relevant change"? Is this the only change affecting user of the API?

asoldano replied:
Right, changes impacting the user applications or requiring a modification of the spec implementation.

honza-kasik commented on microprofile/WFLY-13089-microprofile-rest-client-1.4.adoc:
```diff
@@ -0,0 +1,54 @@
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
```
JFYI: The URL is now being redirected to https://issues.redhat.com/browse

asoldano replied:
Right, I copy-pasted the headers section from an old RFE...

honza-kasik commented on microprofile/WFLY-13089-microprofile-rest-client-1.4.adoc:
```diff
@@ -0,0 +1,54 @@
+
+== Community Documentation
+
+Documentations should be coming as part of the RESTEasy component documentation.
```
The community documentation will be available only as a part of RESTEasy project or as a part of WildFly project?

asoldano replied:
The RESTEasy team will only update the RESTEasy project community documentation.

honza-kasik replied:
Thanks! Can you please include this fact into the document (without "should")?

asoldano replied:
Done

honza-kasik commented on microprofile/WFLY-13089-microprofile-rest-client-1.4.adoc:
```diff
@@ -0,0 +1,54 @@
+
+== Requirements
+
+* RESTEasy provides an implementation of Microprofile REST Client 1.4.0 version
```
This means there will be version bump of RESTEasy in WildFly or are there any other requirements?

asoldano replied:
I don't expect any additional change to be needed in WildFly besides a RESTEasy component upgrade.

asoldano commented:
> Thank you @asoldano for this proposal. I have left some questions behind. Can you also please add a Testing by section (set to QE since MicroProfile has high priority)?
> 
> ```
> === Testing By
> 
> [ ] Engineering
> 
> [x] QE
> ```

Thanks, I've added the "Testing by" section and addressed your other comments, please let me know if there's anything else that needs to be clarified.
